docs/design/block-design.md

```diff
@@ -1,17 +1,88 @@
-- The “Block” tab of the Settings Sidebar can contain additional block options and configuration, but keep in mind that a user might dismiss the sidebar and never use it. Do not put critical options there.
-- Check how your block looks, feels, and works on all sorts of devices and screen sizes.
+### The primary interface for a block is the content area of the block
+Since the block itself represents what will actually appear on the site, interaction here hews closest to the principle of direct manipulation and will be most intuitive to the user. This should be thought of as the primary interface for adding and manipulating content and adjusting how it is displayed. There are two ways of interacting here:
+
+1. The placeholder content in the content area of the block can be thought of as a guide or interface for users to follow a set of instructions or “fill in the blanks”. For example, a block that embeds content from a 3rd-party service might contain controls for signing in to that service in the placeholder.
+2. After the user has added content, selecting the block can reveal additional controls to adjust or edit that content. For example, a signup block might reveal a control for hiding/showing subscriber count. However, this should be done in minimal ways, so as to avoid dramatically changing the size and display of a block when a user selects it (this could be disorienting or annoying).
+
+### The block toolbar is a secondary place for required options & controls
+Basic block settings won’t always make sense in the context of the placeholder / content UI. As a secondary option, options that are critical to the functionality of a block can live in the block toolbar. The block toolbar is still highly contextual and visible on all screen sizes. One notable constraint with the block toolbar is that it is icon-based UI, so any controls that live in the block toolbar need to be ones that can effectively be communicated via an icon or icon group.
+
+### The block sidebar should only be used for advanced, tertiary controls.
+The sidebar is not visible by default on a small / mobile screen, and may also be collapsed in a desktop view. Therefore, it should not be relied on for anything that is necessary for the basic operation of the block. Pick good defaults, make important actions available in the block toolbar, and think of the sidebar as something that only power users may discover. In addition, use sections and headers in the block sidebar if there are more than a handful of options, in order to allow users to easily scan and understand the options available.
+
+
+
+## Do's and Don'ts 
+
+### Blocks
+
+A block should have a straightforward, short name so users can easily find it in the Block Library. A block named "YouTube" is easy to find and understand. The same block, named "Embedded Video (YouTube)", would be less clear and harder to find in the Block Library.
```

Comment:
```suggestion
A block should have a straightforward, short name so users can easily find it in the Block Library. A block named "YouTube" is easy to find and understand. The same block, named "Embedded Video (YouTube)", would be less clear and harder to find in the Block Library. (You can, however, use _keywords_ to make a block searchable with related words. In this case, "embed" and "video" may be appropriate keywords to improve discoverability, while keeping the block name simple.)
```

Reply:
I'm slightly nervous about suggesting keywords because in practice I don't feel they've worked super well, and I'm even more concerned about mis-use, even though we have a 3 keyword limit. For example right now "Image" is also a keyword for "Gallery", and since the slash command doesn't yet have any smartness to it, it means if I just inserted a gallery then next time I type `/image` and just press Enter, I get a gallery.